src/library_async.js

```diff
@@ -424,20 +424,16 @@ mergeInto(LibraryManager.library, {
     },
   },
 
-  // TODO: Update all other methods in this file to support ASYNCIFY=2 / stack
-  //       switching. That requires (1) adding a sig for them (as we need the
-  //       signature to declare the type for the Suspender API), and (2) adding
-  //       a return to their bodies, even if they do not return a value, as the
-  //       Suspender API expects them to return a Promise.
   emscripten_sleep__sig: 'vi',
   emscripten_sleep__deps: ['$safeSetTimeout'],
   emscripten_sleep: function(ms) {
     return Asyncify.handleSleep((wakeUp) => safeSetTimeout(wakeUp, ms));
   },
 
+  emscripten_wget__sig: 'vpp',
   emscripten_wget__deps: ['$Browser', '$PATH_FS', '$FS'],
   emscripten_wget: function(url, file) {
-    Asyncify.handleSleep((wakeUp) => {
+    return Asyncify.handleSleep((wakeUp) => {
```

Comment:
so `ASYNCIFY=2` requires a return whereas =1 didn't?

Seems odd that there is a return statement but function is being declared as void.  I guess it matches the C signature though and this return is not visible of the native code right?  (since its a Promise I guess?)

Reply:
Yeah, it's the Promise. You're right that it's odd. I'll add a comment.